Services/Init/classes/class.ilStartUpGUI.php

```diff
@@ -1517,10 +1548,62 @@ protected function getAcceptance()
         $this->showTermsOfService();
     }
 
+    protected function confirmAcceptance()
+    {
+        $this->showTermsOfService(true);
+    }
+
+    // Show terms of service and modal to confirm withdrawal
+    protected function confirmWithdrawal()
```

Comment:
Is it possible for factor this out to some `TermsOfService` GUI/Rendering class?